src/py/flwr/server/app.py

```diff
@@ -280,25 +304,40 @@ def run_server() -> None:
         state_factory=state_factory,
     )
 
+    grpc_servers = [driver_server]
+    bckg_threads = []
+
     # Start Fleet API
-    fleet_server: grpc.Server = _run_fleet_api_grpc_bidi(
-        address=args.fleet_api_address,
-        state_factory=state_factory,
-    )
+    if args.fleet_api_type == "rest":
+        fleet_thread = threading.Thread(
+            target=_run_fleet_api_rest,
+            args=(args.rest_fleet_api_address, state_factory),
+        )
+        fleet_thread.start()
+        bckg_threads.append(fleet_thread)
+    elif args.fleet_api_type == "grpc":
+        fleet_server = _run_fleet_api_grpc_bidi(
+            address=args.grpc_fleet_api_address,
+            state_factory=state_factory,
+        )
+        grpc_servers.append(fleet_server)
+    else:
+        raise ValueError(f"Unknown fleet_api_type: {args.fleet_api_type}")
 
     # Graceful shutdown
     _register_exit_handlers(
-        grpc_servers=[driver_server, fleet_server],
+        grpc_servers=grpc_servers,
+        bckg_threads=bckg_threads,
         event_type=EventType.RUN_SERVER_LEAVE,
     )
 
     # Block
     driver_server.wait_for_termination()
-    fleet_server.wait_for_termination()
 
 
 def _register_exit_handlers(
     grpc_servers: List[grpc.Server],
+    bckg_threads: List[threading.Thread],
```

Comment:
What does `bckg_threads` mean? Can we rename this?

Reply:
It's a list of running background threads. We can rename it in an update, the solution isn't ideal and might change anyways.

Reply:
✅